tests/_test_utils/test_verify_waypoint_data.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+"""
+
+    tests._test_utils.test_verify_xml_waypoint
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    This tests for valid xml data of waypoint data.
+
+    This file is part of MSS.
+
+    :copyright: Copyright 2024 Reimar Bauer
+    :license: APACHE-2.0, see LICENSE for details.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+from mslib.utils.verify_waypoint_data import verify_waypoint_data
+
+
+def test_verify_xml_waypoint():
```

Comment:
Do we need to test this specifically or can we cover the entirety of that function by only interacting with MSColab's public interface?

Reply:
We have more than one interface. I want to be sure that this is covered when a refactoring is done and whatever sends us data.

Reply:
> We have more than one interface.

And all of them that interact with the real world have to be tested. The important part is that the servers interface to the outside world does indeed do the validation, not that the validation in itself exists and responds as expected.

I just want to emphasize that a test is meant to codify an expected behavior of the code, and the existence of an internal function called `verify_xml_waypoint` is not a crucial part to ensure for the future, in my opinion.

Ensuring that coverage doesn't decrease through a refactoring could be accomplished by checking that coverage does not decrease, instead.

> whatever sends us data

That's the point, this test does not concern itself with if data send from whereever is actually validated.

Reply:
I disagree, I first want to know what kind of failures we are aware off. I also want to know that a new function has no failure when it is designed. For me testdriven means to define what the function should solve.  The test describes what I had in mind as I wrote the function.

Reply:
I am not sure whether this should be discussed on an open channel. However, the XML is never parsed by the server, so the server should be "safe".
The clients are a different matter though. If we think that XML cannot be trusted, simply switching to defusedxml on the client side should solve this...?

Reply:
@joernu76 I think you wanted to respond to https://github.com/Open-MSS/MSS/pull/2468#discussion_r1721359848. Yes, you are right, we should take that to a closed channel.

Reply:
> For me testdriven means to define what the function should solve.

I see that differently. Test-driven means writing tests for the correctness of the change that is to be implemented, in this case for the change that the server and client should verify the xml data. That in the end this function implements the change is an irrelevant implementation detail, as far as testing is concerned IMO. It could be a thousand different functions, and only a single test, as long as the test exercises the change they implement from the codes exposed surface.

Anyway, no matter if this test is present or not, the client-side validation isn't tested for yet. I.e. I can remove the client-side validation without a test failing.